gum/backend-darwin/gumobjcapiresolver.c

```diff
@@ -343,23 +368,35 @@ gum_objc_api_resolver_find_method_by_address (GumApiResolver * resolver,
                                               GumAddress address)
 {
   GumObjcApiResolver * self = GUM_OBJC_API_RESOLVER (resolver);
-  GHashTableIter iter;
-  GumObjcClassMetadata * klass;
+  gchar * result = NULL;
+  gint class_count, class_index;
+  Class * classes;
 
-  g_hash_table_iter_init (&iter, self->class_by_handle);
+  g_rec_mutex_lock (&self->monitor->mutex);
+
+  class_count = self->objc_getClassList (NULL, 0);
+  classes = g_malloc (class_count * sizeof (Class));
```

Comment:
```suggestion
  classes = g_malloc (class_count * sizeof (Class));
```

Would be good to use `g_new()` here.